### ISSUE TITLE
<valarray>: P1614R2 added operator== for slice, but not gslice

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#1676]
[P1614R2](https://wg21.link/P1614R2) Adding Spaceship `<=>` To The Library added `operator==` to `<valarray>`'s `slice`, which contains `size_t start(), size(), stride()` (see WG21-N4878 [class.slice.overview]).

`gslice`, which contains `size_t start()` and `valarray<size_t> size(), stride()` (see N4878 [class.gslice.overview]), wasn't mentioned in P1614R2. As the `valarray` members are equality comparable, it would seem consistent and reasonable to add an equality operator to `gslice`. This seems like an oversight (or if it was intentional, an explanation was missing).

This would be an easy Library Evolution issue to file, or even a small proposal for someone to write.

Originally noticed by @statementreply.

**Comment by timsong-cpp** [comment]
> the `valarray` members are equality comparable

I don't think this is true. Comparing two `valarray`s gives you a `valarray<bool>` (or an expression template that converts to one), not `bool`.

**Comment by StephanTLavavej** [comment]
Thanks, I'll buy that.
